app/js/app-menu.js

```diff
@@ -964,7 +971,133 @@ module.exports = function() {
       var cy = chiseInstance.getCy();
 
       chiseInstance.highlightProcesses(cy.nodes(':selected'));
+      });
+    // PD map to AF map conversion 
+    $("#highlight-errors-of-pd2af").click(function (e) {
+    
+      var chiseInstance = appUtilities.getActiveChiseInstance();
+      var chiseSpinner = appUtilities.getActiveChiseInstance();
+      var filename = document.getElementById('file-name').innerHTML;
+      var fExt = 'sbgn';
+      filename = filename.substring(0, filename.lastIndexOf('.')).concat(".").concat(fExt);
+
+      // Get the map from the canvas
+      
+      var cy = chiseInstance.getCy();
+      var nodes = cy.nodes().filter( function( node ) {
+        // return !chiseInstance.elementUtilities.isSIFNode( node );
+        // console.log(node.visible());
+        return node.visible();
+      } );
+      
+      var edges = cy.edges().filter( function( edge ) {
+        return edge.visible();
+      } );
+      
+      // console.log(nodes);
+      // console.log(edges);
+      var renderInfo = appUtilities.getAllStyles(cy, nodes, edges);
+
+      var properties = appUtilities.getScratch(appUtilities.getActiveCy(), 'currentGeneralProperties');
+
+      // var dene = file.createSbgnml(filename, "plain", renderInfo, properties, nodes, edges);
+      var file = chiseInstance.getSbgnvizInstance().convertSbgn(filename, "plain", renderInfo, properties, nodes, edges);
+      
+      // If the map type is not PD or canvas is empty display error  
+      if(chiseInstance.getMapType() != 'PD'){
+        promptInvalidTypeWarning.render();
+        return;
+      }
+
+      if(nodes.length == 0 || edges.length == 0){
+        promtErrorPD2AF.render("No visible map found!");
+        return;
+      }
+
+      chiseSpinner.startSpinner("layout-spinner");
+
+      var url = "/";
+      var retdata;
+      // pd2af returns filename and file url
+      $.ajax({
+        // After deploying Bridge Server (pd2af-webservice) write the bridge server's URL but leave the /convert
+        url: "https://pd2afwebservice.herokuapp.com/convert",
```

Comment:
Don't forget to change here after you make the service active in the public server.

Reply:
This section is unnecessary. I'll delete it on the next commit

Reply:
Changed to http://localhost:4000/convert